From 17b5815c67ec54b01a86d7cfd4fa09c238905f58 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Mon, 11 Jan 2016 01:36:39 +0200
Subject: [PATCH 045/192] Snap: Fall back to default quality for invalid video
 qualities

Might happen when updating from previous versions of Snap

Change-Id: I42d2037b4e32c7f5c88a60b5da1895dcb761d176
---
 src/com/android/camera/VideoModule.java | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 38be63964..a04dc3027 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -872,9 +872,8 @@ public class VideoModule implements CameraModule,
     private void readVideoPreferences() {
         // The preference stores values from ListPreference and is thus string type for all values.
         // We need to convert it to int manually.
-        String videoQuality = mPreferences.getString(CameraSettings.KEY_VIDEO_QUALITY,
-                        null);
-        if (videoQuality == null) {
+        String videoQuality = mPreferences.getString(CameraSettings.KEY_VIDEO_QUALITY, null);
+        if (videoQuality == null || (videoQuality.length() < 3 && !videoQuality.contains("x"))) {
             mParameters = mCameraDevice.getParameters();
             String defaultQuality = mActivity.getResources().getString(
                     R.string.pref_video_quality_default);
-- 
2.17.1

